MongoConsumerLibary/MongoConnection/MongoConnection.cs

```diff
@@ -0,0 +1,37 @@
+﻿using MongoConsumerLibary.MongoConnection.Collections;
+using MongoConsumerLibary.MongoConnection.Enums;
+using MongoDB.Driver;
+using System;
+using System.Collections.Generic;
+
+namespace MongoConsumerLibary.MongoConnection
+{
+    public class MongoConnection
+    {
+        private readonly MongoClient _mongoClient;
+        private readonly IMongoDatabase _database;
+        private readonly Dictionary<CollectionType, IMongoCollection<BaseBoxCollection>> _collections;
+        private readonly MongoSettings _mongoSettings;
+        public MongoConnection(MongoSettings mongoSettings)
+        {
+            _mongoSettings = mongoSettings;
+            _mongoClient = new MongoClient(_mongoSettings.ConnectionUrl);
+            _database = _mongoClient.GetDatabase(_mongoSettings.DataBaseName);
+            _collections = new Dictionary<CollectionType, IMongoCollection<BaseBoxCollection>>();
```

Comment:
You don't need collection for every IcdType, put in one collection and add icdType as a property